### PR TITLE
chore(18780): refactor get test result by system number and staff id

### PR DESCRIPTION
**Pull request by m-ujaffer**
get test results by system number and get test results by testerStaffId refactor.

https://jira.dvsacloud.uk/browse/CVSB-18780
https://jira.dvsacloud.uk/browse/CVSB-18781

_NOTE: This PR is configured with base as feature/CVSB-18734._

### CODE REVIEW

File: src/handlers/expiry/providers/TestDataProvider.ts [hunk 1]
```diff
@@ -1,25 +1,57 @@
...
     const fromDateTime = new Date(1970, 1, 1);
     const toDateTime = new Date();
-    let result: ITestResult[] = [];
+    let result: models.ITestResult[] = [];
     try {
-      this.testResultsDAO = this.testResultsDAO as TestResultsDAO;
+      this.testResultsDAO = this.testResultsDAO as models.TestResultsDAO;
       const data = await this.testResultsDAO.getBySystemNumber(systemNumber);
       console.log(`getTestHistory: Data Count -> ${data?.Count}`);
       if (data?.Count) {
```
Comment (bristol-mike-jones) [inline src/handlers/expiry/providers/TestDataProvider.ts:53]: We can flatten this out with a guard that returns if there are no results

File: src/handlers/expiry/providers/TestDataProvider.ts [hunk 2]
```diff
@@ -1,25 +1,57 @@
...
-        result = data.Items as ITestResult[];
+        result = data.Items as models.ITestResult[];
         return result
```
Comment (bristol-mike-jones) [inline src/handlers/expiry/providers/TestDataProvider.ts:55]: This can be flattened out too

File: src/handlers/expiry/providers/TestDataProvider.ts [hunk 3]
```diff
@@ -1,25 +1,57 @@
-import { HTTPError } from "../../../models/HTTPError";
+import moment from "moment";
```
Comment (bristol-mike-jones) [inline src/handlers/expiry/providers/TestDataProvider.ts:1]: Do we need moment in here? We are trying to constrain moment to an internal of the date utils

File: src/handlers/expiry/providers/TestDataProvider.ts [hunk 4]
```diff
@@ -95,4 +124,89 @@ export class TestDataProvider implements ITestDataProvider {
...
+    return testResults.filter((testResult) => {
+      const k = filterName as keyof typeof testResult;
+      return testResult[k] === filterValue;
+    });
+  }
+
+  public static filterTestResultByDate(testResults: models.ITestResult[], fromDateTime: string | number | Date, toDateTime: string | number | Date): models.ITestResult[] {
+
+    return testResults.filter((testResult: { testStartTimestamp: string | number | Date; testEndTimestamp: string | number | Date; }) => {
+      return moment(testResult.testStartTimestamp).isAfter(fromDateTime) && moment(testResult.testEndTimestamp).isBefore(toDateTime);
```
Comment (bristol-mike-jones) [inline src/handlers/expiry/providers/TestDataProvider.ts:162]: Can we push this into the date utils so that this class doesn't need to use moment.  Keeps it cleaner.

**Review by bristol-mike-jones — CHANGES_REQUESTED** [review]

File: src/handlers/expiry/providers/TestDataProvider.ts (hunk 3 above)
Comment (m-ujaffer) [reply src/handlers/expiry/providers/TestDataProvider.ts:1]: done

File: src/handlers/expiry/providers/TestDataProvider.ts (hunk 1 above)
Comment (m-ujaffer) [reply src/handlers/expiry/providers/TestDataProvider.ts:53]: done

File: src/handlers/expiry/providers/TestDataProvider.ts (hunk 2 above)
Comment (m-ujaffer) [reply src/handlers/expiry/providers/TestDataProvider.ts:55]: done

File: src/handlers/expiry/providers/TestDataProvider.ts (hunk 4 above)
Comment (m-ujaffer) [reply src/handlers/expiry/providers/TestDataProvider.ts:162]: done

File: src/services/TestResultsService.ts [hunk 5]
```diff
@@ -27,71 +26,20 @@ export class TestResultsService {
...
-            console.error("dynamoError on getBySystemNumber", error);
-            error = new models.HTTPError(500, enums.MESSAGES.INTERNAL_SERVER_ERROR);
-          }
-          throw error;
-        }
-      } else {
-        // console.log("invalid systemNumber Filters", filters);
-        return Promise.reject(new models.HTTPError(400, enums.MESSAGES.BAD_REQUEST));
-      }
+  public async getTestResultBySystemNumber(filters: models.ITestResultFilters): Promise<models.ITestResult[]> {
```
Comment (JonLazariniDD) [inline src/services/TestResultsService.ts:29]: 👍

File: src/services/TestResultsService.ts [hunk 6]
```diff
@@ -524,4 +472,13 @@ export class TestResultsService {
       });
     return payload;
   }
+
+  private static handleError(error: any) {
+    console.error("TestResultsService.getTestResultBySystemNumber: error -> ", error);
+    const httpError = error as models.HTTPError;
+    if (!(httpError && [400, 404].includes(httpError.statusCode))) {
```
Comment (JonLazariniDD) [inline src/services/TestResultsService.ts:479]: I will need a catch up regarding that please.

**Review by JonLazariniDD — APPROVED** [review]

File: src/handlers/expiry/providers/TestDataProvider.ts [hunk 7]
```diff
@@ -1,54 +1,104 @@
...
+   */
+  public async getTestResultBySystemNumber(
+    filters: models.ITestResultFilters
+  ): Promise<models.ITestResult[]> {
+    try {
+      this.testResultsDAO = this.testResultsDAO as models.TestResultsDAO;
+      const result = await this.testResultsDAO.getBySystemNumber(
+        filters.systemNumber
+      );
+      const response: models.ITestResultData = {
```
[the remark this answers is not in the document]
Comment (m-ujaffer) [reply src/handlers/expiry/providers/TestDataProvider.ts:24]: updated

File: src/utils/GetTestResults.ts [hunk 8]
```diff
@@ -12,13 +12,6 @@ export class GetTestResults {
     });
   }
 
-  public static filterTestResultByDate(testResults: any, fromDateTime: string | number | Date, toDateTime: string | number | Date) {
-
-    return testResults.filter((testResult: { testStartTimestamp: string | number | Date; testEndTimestamp: string | number | Date; }) => {
-      return moment(testResult.testStartTimestamp).isAfter(fromDateTime) && moment(testResult.testEndTimestamp).isBefore(toDateTime);
```
Comment (bristol-mike-jones) [inline src/utils/GetTestResults.ts:18]: Tidy the imports. moment is no longer used.

Comment (m-ujaffer) [reply src/utils/GetTestResults.ts:18]: The changes requested were minor so they have been pushed to 18782 to reduce rebases.

**Review by bristol-mike-jones — APPROVED** [review]

File: src/handlers/expiry/providers/DateProvider.ts [hunk 9]
```diff
@@ -100,8 +100,8 @@ export class DateProvider {
   public static isBetweenDates(
     fromDate: string | number | Date,
     toDate: string | number | Date,
-    compareFromDate: Date,
-    compareToDate: Date
+    compareFromDate: string | number | Date,
+    compareToDate: string | number | Date
```
Comment (alex-deloitte) [inline src/handlers/expiry/providers/DateProvider.ts:104]: Why did you need to allow storing dates as numbers here? I would avoid it

File: src/services/TestResultsService.ts (hunk 5 above)
Comment (alex-deloitte) [inline src/services/TestResultsService.ts:29]: I wouldn't spend too much refactoring the old code in this service since it will be deprecated. Actually, shouldn't you mark these functions as deprecated?

File: src/handlers/expiry/providers/DateProvider.ts (hunk 9 above)
Comment (m-ujaffer) [reply src/handlers/expiry/providers/DateProvider.ts:104]: All numbers have been removed from DateProvider in 18783.

File: src/services/TestResultsService.ts (hunk 5 above)
Comment (m-ujaffer) [reply src/services/TestResultsService.ts:29]: This function may still be used for VTM.

**Review by alex-deloitte — APPROVED** [review]
